master/buildbot/process/buildstep.py

```diff
@@ -1008,8 +1010,8 @@ def checkDisconnect(self, f):
 
     def commandComplete(self, cmd):
         # Now that the build is complete, add any URLs that the buildstep had to the buildstep's status
-        for url_key in self.urls:
-            self.addURL(url_key, self.urls[url_key])
+        for url in self.urls:
+            self.addURL(url["name"], url["url"])
```

Comment:
It's looks like unnecessary casting. Maybe it's better just append all urls list without extracting and casting?